### ISSUE TITLE
Don't shade packetevents anymore

### ISSUE THREAD

**Issue by ghost** [GrimAnticheat/Grim#1528]
### Feature description

Pros of unshaded packetevents:

- Increased performance, multiple plugins using packet events can share the same instance.
- Allows server owners to update packet events without waiting on grim to do it.
- Halves grims jar size
- Seperation of concerns
- @retrooper gets credit for his work (Downloads for shaded packet events aren't counted, if unshaded he gets downloads/recognition/ad revenue)

Cons:
- Server owners need to download 2 jars, one for packet events one for grim

**Comment by SamB440** [comment]
No, we need control over the listener priorities and to have our own encoder/decoder, you will get conflicts with other plugins otherwise

**Comment by ghost** [comment]
> No, we need control over the listener priorities and to have our own encoder/decoder, you will get conflicts with other plugins otherwise

I don't understand, you shade packet events directly from the github repo, all you do is relocate it?

**Comment by SamB440** [comment]
Shading packetevents gives Grim its own encoder/decoder, this also means we don't have any plugins interfering in packet listener priorities, and any plugins that mess with packets using packetevents (such as spawning entities) will go through Grim.

**Comment by ghost** [comment]
> Shading packetevents gives Grim its own encoder/decoder, this also means we don't have any plugins interfering in packet listener priorities, and any plugins that mess with packets using packetevents (such as spawning entities) will go through Grim.

Seems like a non issue, why would plugin be messing with spawning entity packets? Grim doesnt modify any packets right? just reading? For packets we are receiving grim should listen at lowest priority, for packets being sent listen at highest, simple no?

**Comment by SamB440** [comment]
No, I suggest you look at all the different priorities Grim is using. I have packet-level NPC plugins, if Grim was not shaded then my packets might not be read by Grim, and other plugins may listen at different packet priorities and modify packets.

**Comment by AoElite** [comment]
@IssueDescription Retrooper is fine with people shading packetevents & he isn't getting ad revenue anyways.

**Comment by retrooper** [comment]
Modrinth does offer a little ad-revenue. At the end of the day, *not* shading benefits the users so it has little to do with me.

**Comment by Axionize** [comment]
This has been resolved with the new shadePE build flag to optionally build Grim without PE shaded which will rely on external PE. Note that you will need to use our fork of PE for Grim to function properly.

https://github.com/GrimAnticheat/packetevents
